<launch>
  <node pkg="collision_map" type="collision_map_occ_node" name="collision_map_occ" respawn="true" output="screen">
    <remap from="cloud_in" to="full_cloud_annotated" />
    <remap from="cloud_incremental_in" to="tilt_scan_cloud_annotated" />

    <remap from="robot_description" to="robotdesc/pr2" />
    
    <param name="publish_occlusion" type="bool" value="true" />

    <!-- tell the collision map there may be an annotation channel telling which points are on the robot -->
    <param name="cloud_annotation" type="string" value="on_self" />

    <!-- a frame that does not change as the robot moves -->
<!--    <param name="fixed_frame" type="string" value="/odom_combined" /> -->
    <param name="fixed_frame" type="string" value="base_link" />

    <!-- define a box of size 2x3x4 around (1.1,  0, 0) in the robot frame -->
    <param name="robot_frame" type="string" value="base_link" />

    <param name="origin_x" type="double" value="1.1" />
    <param name="origin_y" type="double" value="0.0" />
    <param name="origin_z" type="double" value="0.0" />
    
    <param name="dimension_x" type="double" value="1.0" />
    <param name="dimension_y" type="double" value="1.5" />
    <param name="dimension_z" type="double" value="2.0" />

    <!-- set the resolution (1.5 cm) -->
    <param name="resolution" type="double" value="0.015" />

    <!-- the amount of time to remember an occluded point for -->
    <param name="keep_occluded" type="double" value="30.0" />

    <!-- when occluded obstacles are raytraced, keep boxes from occluded space within a given radius -->
    <param name="radius" type="int" value="5" />

    <!-- define sensor location in robot frame -->
    <param name="sensor_x" type="double" value="0.05" />
    <param name="sensor_y" type="double" value="0.0" />
    <param name="sensor_z" type="double" value="1.0" />

  </node>
</launch>
